telethon_examples/auto_reply.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+# disclaimer: you should not actually use this. it can be quite spammy.
+from time import sleep
+from telethon import TelegramClient
+from getpass import getpass
+from telethon.tl.types import InputPeerUser,InputPeerChannel
+from telethon.tl.types import Updates
+from telethon.tl.types import UpdateNewChannelMessage,UpdateNewMessage
+from telethon.tl.functions.messages import SendMessageRequest,EditMessageRequest
+from telethon.tl.types import MessageService
+from nltk.tokenize import word_tokenize
+from os import environ
+
+CHANNELS = {}
+CHANNELNAMES = {}
+USERS = {}
+EMACS_BLACKLIST = [1058260578, # si @linux_group
+        123456789]
+REACTS = {'emacs':'Needs more vim.',
+        'chrome':'Needs more firefox.',
+}
+
+class NeedsMore(TelegramClient):
+    def __init__(self):
+        settings = {'api_id':int(environ['TG_API_ID']),
+                'api_hash':environ['TG_API_HASH'],
+                'user_phone':environ['TG_PHONE'],
+                'session_name':'needsmore'}
+        super().__init__(
+            settings.get('session_name','session1'),
+            settings['api_id'],
+            settings['api_hash'],
+            proxy=None,
+            process_updates=True)
+
+        user_phone = settings['user_phone']
+
+        print('INFO: Connecting to Telegram Servers...', end='', flush=True)
+        self.connect()
+        print('Done!')
+
+        if not self.is_user_authorized():
+            print('INFO: Unauthorized user')
+            self.send_code_request(user_phone)
+            code_ok = False
+            while not code_ok:
+                code = input('Enter the auth code: ')
+                try:
+                    code_ok = self.sign_in(user_phone, code)
+                except RPCError as e:
+                    if e.password_required:
+                        pw = getpass('Two step verification enabled. Please enter your password: ')
+                        self.sign_in(password=pw)
+                    else:
+                        raise e
```

Comment:
You can just write `raise` and it will work. No need to do so once you change `RPCError` for `SessionPasswordNeededError` though.